### PR TITLE
docs: how to write reousrce-controller

### PR DESCRIPTION
**Pull request by xiao-jay**
docs about how to write reousrce-controller

### CODE REVIEW

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Resource-controller如何编写
+
+resource-controller 是计量计费模块的一部分，请确保您提前了解[计量计费的设计](https://github.com/labring/sealos/blob/main/proposal/design/zh/metering%E8%AE%BE%E8%AE%A1.md)
```
Comment (zzjin) [inline docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md:3]: use markdown link to relative url like [](/proposal/design/zh/metering%E8%AE%BE%E8%AE%A1.md)

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+
+
+## resource-controller 是什么
+
+resource-controller 是计量计费模块的一部分，用来统计和计费资源的使用量，不同资源不同resource-controller，使用podresource-controller举例，podresource-controller会统计用户ns下pod的资源limit使用量，并且计算出价格，生成一个Resource CR.
+
+![](https://p.ipic.vip/p87a3h.png)
```
Comment (zzjin) [inline docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md:11]: replace with repo url

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	LatestUpdateTime int64 `json:"latestUpdateTime,omitempty"`
+	SeqID            int64 `json:"seqID,omitempty"`
+}
+```
+
+seqID是每次podresource-controller统计计费完之后都需要+1，用当前seqID值来作为后缀命名创建的Resource CR，防止CR 名字一样。
+
+### 在reconcile 中实现以下接口
+
+https://github.com/labring/sealos/blob/main/pkg/metering/interface.go
```
Comment (zzjin) [inline docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md:40]: replace with relative file path

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md (hunk 1 above)
Comment (xiao-jay) [reply docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md:3]: linke this ? ](../../../../../proposal/design/zh/metering%E8%AE%BE%E8%AE%A1.md)

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md (hunk 3 above)
Comment (xiao-jay) [reply docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md:40]: (../../../../../pkg/metering/interface.go) is ok?

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md (hunk 2 above)
Comment (xiao-jay) [reply docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md:11]: is ok?

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md (hunk 1 above)
Comment (zzjin) [reply docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md:3]: > linke this ? (../../../../../proposal/design/zh/metering%E8%AE%BE%E8%AE%A1.md)

Just `/proposal/design/zh/metering%E8%AE%BE%E8%AE%A1.md` is enough?

File: docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+resource-controller 是计量计费模块的一部分，请确保您提前了解[计量计费的设计](/proposal/design/zh/metering%E8%AE%BE%E8%AE%A1.md)
+
+
+
+## resource-controller 是什么
+
+resource-controller 是计量计费模块的一部分，用来统计和计费资源的使用量，不同资源不同resource-controller，使用podresource-controller举例，podresource-controller会统计用户ns下pod的资源limit使用量，并且计算出价格，生成一个Resource CR.
+
+<img src="/proposal/img/metering-proposal-3.png" style="zoom:33%;" />
```
[the remark this answers is not in the document]
Comment (xiao-jay) [reply docs/4.0/i18n/zh-Hans/Development-Guidelines/resource-controller.md]: done

**Pull request merged**